Examples/Detectors/TGeoDetector/include/ActsExamples/TGeoDetector/TGeoDetector.hpp

```diff
@@ -61,8 +61,7 @@ struct TGeoDetector {
         Acts::TGeoLayerBuilder::defaultElementFactory;
 
     /// Optional geometry identfier hook to be used during closure
-    /// @note Will be @b copied when calling the geometry building
-    Acts::GeometryIdentifierHook geometryIdentifierHook;
+    std::shared_ptr<const Acts::GeometryIdentifierHook> geometryIdentifierHook;
```

Comment:
Default also here or not needed?

Reply:
Good catch, should be added to prevent crashes from default constructed configs.